msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/ResetPasswordStates.kt

```diff
@@ -256,9 +280,15 @@ class ResetPasswordCodeRequiredState internal constructor(
 class ResetPasswordPasswordRequiredState internal constructor(
     override val flowToken: String,
     private val config: NativeAuthPublicClientApplicationConfiguration
-) : BaseState(flowToken), State, Serializable {
+) : BaseState(flowToken), State, Parcelable {
     private val TAG: String = ResetPasswordPasswordRequiredState::class.java.simpleName
 
+    constructor(parcel: Parcel) : this(
+        parcel.readString()!!,
+        parcel.readSerializable() as NativeAuthPublicClientApplicationConfiguration
```

Comment:
This is inconsistent with other instances of this constructor. E.g. in the Parcel constructor in `ResetPasswordCodeRequiredState` you're not reading (and setting) the configuration object.

Reply:
We should not write and read config objects as it could lead to stale configuration.

Reply:
Could we write like this if NativeAuthPublicClientApplicationConfiguration is not a singleton pattern?

```
constructor(parcel: Parcel) : this(
        parcel.readString() ?: "",
        NativeAuthPublicClientApplicationConfiguration()
}
```

Reply:
Done